File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/local/projects/l3-resource-keyword-overlap/keywordComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_simple as simple

class KeywordComponentArgs(TypedDict, total=False):
    input: Input[bool]

class KeywordComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: KeywordComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:KeywordComponent", name, args, opts)

        # A resource named `this` collides with the receiver pointer of the
        # ComponentResource class generated for this component. NodeJS must rename the
        # resource variable (e.g. to `_this`) while keeping the `parent: this` pointer
        # intact.
        this = simple.Resource(f"{name}-this", value=args["input"],
        opts = pulumi.ResourceOptions(parent=self))

        # Referencing `this` exercises that the rename is applied to references too, not
        # just the declaration. The name `parent` also overlaps with the `parent`
        # resource-option key, which must not be confused with this resource variable.
        parent = simple.Resource(f"{name}-parent", value=this.value,
        opts = pulumi.ResourceOptions(parent=self))

        self.result = parent.value
        self.register_outputs({
            'result': parent.value
        })

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-resource-keyword-overlap/keywordComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as simple from "@pulumi/simple";

interface KeywordComponentArgs {
    /**
     * An input passed to the component
     */
    input: pulumi.Input<boolean>,
}

export class KeywordComponent extends pulumi.ComponentResource {
    public result: pulumi.Output<boolean>;
    constructor(name: string, args: KeywordComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:KeywordComponent", name, args, opts);
        // A resource named `this` collides with the receiver pointer of the
        // ComponentResource class generated for this component. NodeJS must rename the
        // resource variable (e.g. to `_this`) while keeping the `parent: this` pointer
        // intact.
        const _this = new simple.Resource(`${name}-this`, {value: args.input}, {
            parent: this,
        });

        // Referencing `this` exercises that the rename is applied to references too, not
        // just the declaration. The name `parent` also overlaps with the `parent`
        // resource-option key, which must not be confused with this resource variable.
        const parent = new simple.Resource(`${name}-parent`, {value: _this.value}, {
            parent: this,
        });

        this.result = parent.value;
        this.registerOutputs({
            result: parent.value,
        });
    }
}
